ReSwift/CoreTypes/Store.swift

```diff
@@ -85,40 +85,48 @@ open class Store<State: StateType>: StoreType {
         }
     }
 
-    open func subscribe<S: StoreSubscriber>(_ subscriber: S)
-        where S.StoreSubscriberStateType == State {
-            _ = subscribe(subscriber, transform: nil)
-    }
-
-    open func subscribe<SelectedState, S: StoreSubscriber>(
-        _ subscriber: S, transform: ((Subscription<State>) -> Subscription<SelectedState>)?
-    ) where S.StoreSubscriberStateType == SelectedState
+    private func _subscribe<SelectedState, S: StoreSubscriber>(
+        _ subscriber: S, original: Subscription<State>,
+        transformed: Subscription<SelectedState>?)
+        where S.StoreSubscriberStateType == SelectedState
     {
         // If the same subscriber is already registered with the store, replace the existing
         // subscription with the new one.
         if let index = subscriptions.index(where: { $0.subscriber === subscriber }) {
             subscriptions.remove(at: index)
         }
 
-        // Create a subscription for the new subscriber.
-        let originalSubscription = Subscription<State>()
-        // Call the optional transformation closure. This allows callers to modify
-        // the subscription, e.g. in order to subselect parts of the store's state.
-        let transformedSubscription = transform?(originalSubscription)
-
         let subscriptionBox = self.subscriptionBox(
-            originalSubscription: originalSubscription,
-            transformedSubscription: transformedSubscription,
+            originalSubscription: original,
+            transformedSubscription: transformed,
             subscriber: subscriber
```

Comment:
I actually prefer the more explicit argument names. What do you think about reverting this part of the diff?

Reply:
No problem